### PR TITLE
Update requirements pinning

### PR DESCRIPTION
**Pull request by astrojuanlu**
See https://github.com/readthedocs/readthedocs.org/issues/8616#issuecomment-961883441.

### CODE REVIEW

**Review by stsewd — APPROVED** [review]
These are examples, but looks like people are copying/pasting those exact versions

File: docs/guides/reproducible-builds.rst [hunk 1]
```diff
@@ -121,9 +129,9 @@ Some examples:
 
       # File: docs/requirements.txt
 
-      sphinx==3.4.3
-      sphinx_rtd_theme==0.5.1
-      readthedocs-sphinx-search==0.1.0rc3
+      sphinx==4.2.0
+      sphinx_rtd_theme==1.0.1
```
Comment (astrojuanlu) [inline docs/guides/reproducible-builds.rst:133]:
```suggestion
      sphinx_rtd_theme==1.0.0
```

**Comment by astrojuanlu** [comment]
🤖 Rendered version: https://docs--8655.org.readthedocs.build/en/8655/guides/reproducible-builds.html

**Comment by humitos** [comment]
> These are examples, but looks like people are copying/pasting those exact versions

Yeah 😞  --we may need to add a generic note or something. We can't keep these dependencies updated every time a new version is released.

**Comment by humitos** [comment]
@astrojuanlu also, I understand that the author of the original comment wanted to add the pin for `docutils` in particular, not just updating the version's numbers. This PR does not address that.

**Comment by astrojuanlu** [comment]
@humitos There were two options: pinning docutils, or upgrading Sphinx (that in recent times started pinning docutils). I opted for the latter.

**Pull request merged**